Test/NativeUnitTest/NativeUnitTest.cpp

```diff
@@ -1,23 +1,94 @@
 #include "pch.h"
 #include "CppUnitTest.h"
+#include "../Test/CharOccurrences.cpp"
+#include "../Test/ConsoleReader.cpp"
 #include "../Test/Source.cpp"
+#include <isolator.h>
+#include <string>
 
 using namespace Microsoft::VisualStudio::CppUnitTestFramework;
 
 namespace NativeUnitTestForMain
 {
 	TEST_CLASS(MainTests)
 	{
+		CharOccurrences calculateChars;
+		std::string path = "D:/bruhmoment/Abyss/test.txt";
+		int * charOccurs = calculateChars.getCharOccurs();
 	public:
 		
-		TEST_METHOD(TestIfPathExists)
+		TEST_METHOD(TestIfInputPathExists)
 		{
-			int charOccurs[possibleChars];
-			for (int i = 0; i < possibleChars; ++i) {
-				charOccurs[i] = 0;
-			}
-			Assert::IsFalse(calculateOccursInFile("randompath.txt", {}));
-			Assert::IsTrue(calculateOccursInFile("D:/bruhmoment/Abyss/ToDo.txt", charOccurs));
+			Assert::IsTrue(calculateChars.calculateOccursInFile(path, charOccurs));
+		}
+		TEST_METHOD(TestIfOutputPathExists)
+		{
+			Assert::IsTrue(calculateChars.calculateOccursInFile(path, charOccurs));
+		}
+		TEST_METHOD(TestCalculateOccursInLine) 
+		{
+			std::string line = "aaA  ";
+			calculateChars.calculateOccurs(charOccurs, line);
+			Assert::AreEqual(2, charOccurs[int('a')]);
+		}
+		TEST_METHOD(TestConsoleReader)
+		{
+			// ConsoleReader* reader = FAKE<ConsoleReader>();
+			ConsoleReader reader;
+			FAKE_GLOBAL(_getch);
+			char testChars[] = {'a', 'b', reader.backspace, 'c', reader.enter};
+
+
+			// int * timesCalled = TIMES_CALLED(_getch());
+			// WHEN_CALLED(_getch()).Return(testChars[int(timesCalled)]);
+			// why doesn't this work ;(
+
+
+			WHEN_CALLED(_getch()).Return(testChars[0]);
+			WHEN_CALLED(_getch()).Return(testChars[1]);
+			WHEN_CALLED(_getch()).Return(testChars[2]);
+			WHEN_CALLED(_getch()).Return(testChars[3]);
+			WHEN_CALLED(_getch()).Return(testChars[4]);
+			reader.readConsole("testing some letters with backspace and enter");
+			Assert::AreEqual(2, int(reader.getData().length()));
+
+
+			WHEN_CALLED(_getch()).Return(reader.esc);
+			reader.readConsole("testing escape");
+			Assert::AreEqual(0, int(reader.getData().length()));
+		}
+		 TEST_METHOD(TestCalculateOccursInFile)
+		{
+			// you can't fake ifstream with this framework :/
+			 std::ofstream testFile;
+			 std::string testPath = "testFile.txt";
+			 std::string testData = "testData";
+			 testFile.open(testPath);
+			 testFile << testData;
+			 Assert::IsTrue(testFile.is_open());
+			 testFile.close();
+
+			 // int possibleCharsForTest = calculateChars.getPossibleChars();
+			 const int possibleCharsForTest = 128;
```

Comment:
Ouch... Magic number... How dare you?

Reply:
I'm sorry :(

But seriously this was just me giving up on trying to access original possibleChars. Yeah, that's not how one should deal with problems. I guess it works now, with public possibleChars